### PR TITLE
lilypond: Synchronize Portfile with 'lilypond-devel' Portfile.

### PR DESCRIPTION
**Pull request by lemzwerg**
#### Description
Add variant 'mactex'.

Other, minor adjustments.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.7.5 11G63
Xcode 4.6.3 4H1503 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by lemzwerg** [comment]
BTW, the next step will be to merge 'lilypond' and 'lilypond-devel' to one Portfile, providing the latter as a subport.  After this one has been merged, I'll produce another PR.

File: textproc/lilypond/Portfile [hunk 1]
```diff
@@ -59,9 +61,72 @@ build.type          gnu
...
+    "${prefix}/share/texmf-texlive/fonts/opentype/public/tex-gyre"
+set lilypond.mactex_bin   ""
+set lilypond.temp         ""
+set lilypond.have_texgyre false
+
+variant mactex description {Allow lilypond to use MacTeX or another\
+    external TeXLive installation instead of MacPorts TeXLive tools.  To\
+    install the 'mactex' variant the path to the TeX distribution's\
+    binary directory (for example '/Library/TeX/texbin') must be added\
+    to 'binpath' in 'macports.conf' *before* installing this port.} {
```
Comment (mf2k) [inline textproc/lilypond/Portfile:74]: This useful install hint (the 2nd sentence) should be moved into a `notes` block inside this variant.

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: Hmm. My gut feeling says that the message should better stay in the `variant` output, since it is essential information that should be displayed *before* Joe User tries to install it.

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: I don't see an immediate benefit of doubling the data, since the variant part contains checks that test whether MacTeX is actually installed and functioning.

Comment (mf2k) [reply textproc/lilypond/Portfile:74]: My issue is that the variant description never appears when installing the port. One would have to type `port variants lilypond` to see the note at all.

Comment (mf2k) [reply textproc/lilypond/Portfile:74]: hopefully the maintainer will chime in.

**Comment by lemzwerg** [comment]
> My issue is that the variant description never appears when installing the port. One would have to type port variants lilypond to see the note at all.

Yes, and I think this is correct.  If you type `port info lilypond`, you will see that there exists a 'mactex' variant.  If you now type `port variant lilypond`, you will see the warning.  The user is expected to *now* install MacTeX (if he or she hasn't done so already), since `port install lilypond +mactex` will fail otherwise – as mentioned, there are tests for that in the Portfile.  Displaying the hint later is thus completely useless.

Am I missing something?

**Comment by lemzwerg** [comment]
LilyPond 2.21 is out, so I would really like to have this PR handled.  Please have a look!

File: textproc/lilypond/Portfile (hunk 1 above)
Comment (nerdling) [reply textproc/lilypond/Portfile:74]: This is a rather long message for `port variants`. I'd like to see a short message for the variant, with a longer note in `port notes`.

**Comment by nerdling** [comment]
Ryan raised issues with the previous change's pango dependency. Do you plan to make another MR or should that be included in this one?

**Comment by lemzwerg** [comment]
> Ryan raised issues with the previous change's pango dependency [...]
Link, please.  Right now I can't find his remark.

**Comment by nerdling** [comment]
> Link, please. Right now I can't find his remark.

https://github.com/macports/macports-ports/commit/655a3e5efef1390673f3aa6b2c81f28abe824690#r38253922

**Comment by lemzwerg** [comment]
> > Link, please. Right now I can't find his remark.
> 
> [655a3e5#r38253922](https://github.com/macports/macports-ports/commit/655a3e5efef1390673f3aa6b2c81f28abe824690#r38253922)

Thanks; I pinged Jahrme. I think fixing this and other dependency buglets should be a follow-up PR.

**Comment by nerdling** [comment]
Alrighty, as soon as the notes/variants messaging is tweaked we're good to go.

File: textproc/lilypond/Portfile (hunk 1 above)
Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: I still don't get it.  What part of the message should become the 'longer note' that gets shown *after* installation?  All information currently in the variant message is intended to be shown *before* installation!  In comparison to, say, `port notes python37`, no user action has to be done after the package gets installed.

And why is it important to have a short variant message?

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: Can you point me to another macports package that needs a prerequisite (probably with some additional explanation) before a variant can be installed, and that behaves as you envision?

Comment (nerdling) [reply textproc/lilypond/Portfile:74]: Users can invoke `port notes` anytime, just like `port variants`. In practice, one is a list of options and the other is open-ended notes.

I'm fairly certain that most Portfiles with a `prefetch` block in them try to abort builds as early as possible (perhaps due to missing dependencies).

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: > Users can invoke `port notes` anytime, just like `port variants`. In practice, one is a list of options and the other is open-ended notes.

Thanks.  Unfortunately, this doesn't answer my questions.  Let me rephrase it: why shall I put stuff into `port notes` if the relevant information is (a) necessary only for a certain variant, and (b) only *before* installation?  Can you name a package in macports that behaves as you want me to change lilypond?

Comment (nerdling) [reply textproc/lilypond/Portfile:74]: A: it's _a lot_ of information. Too much even. Can you show that it's normal for such a large variant description?

B: `port notes` can be run prior to installation.

Comment (mf2k) [reply textproc/lilypond/Portfile:74]: `notes` are meant for installation information and variant descriptions are not. Most people won't think to look there if they have a problem and will simply file a ticket. Thus it falls in the maintainer's lap.

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: There are some long variant descriptions in macports.  Examples: adblock2privoxy, calendar-contacts-server, macos-fortress.  And none of those packages have the variant information (or a variation of it) repeated in the `port notes` output.

Joe User will normally type `port variants lilypond`, and he will see there are only two variants.  Why is it important to have short and terse descriptions for just two options?  IMHO it's much better to have a detailed explanation than the opposite.

Interestingly, `port info` tells the user that there are variants, but it doesn't tell that there are notes...

And again my really serious questions: What exactly do you expect in the 'notes' section?  And how could the variant information be shortened?  I guess I'm dense, so please help.

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: > `notes` are meant for installation information

This is not what `man portfile` says: "Notes for setup and use of the port. This is shown after the port is activated and anytime the `notes` command is used;" This clearly indicates that the 'notes' section is for information of what the user should do *after* installation.

> Most people won't think to look there if they have a problem and will simply file a ticket. Thus it falls in the maintainer's lap.

This is a good argument, so I could add
```
notes  "This port uses the 'mactex' variant
        and thus needs an external TeX installation."
```

or something similar to the 'mactex' variant block.  Would that work?

However, I still think that the current docstring of `variant` should not be shortened.

Comment (lemzwerg) [reply textproc/lilypond/Portfile:74]: I have to correct myself.  MacTeX is only a build dependency; in other words, after installing LilyPond it is not needed any more, and my suggested wording of a 'notes' block is wrong.  I'm really clueless now what I could write instead.  Please advise.

**Comment by lemzwerg** [comment]
Following your guide I've now added something.  Please check.
